### PR TITLE
fix(custom-resources): AwsCustomResource with delete only action fails

### LINKED ISSUE #6061: AwsCustomResource with delete only action fails with Invalid PhysicalResourceId
I have written some lambda backed custom resources using the provider framework but in some cases the task involved is very simple and does not require the overhead of creating a bespoke lambda.

The CDK offers an alternative to the provider framework which is a custom resource that calls an AWS SDK API
>Sometimes a single API call can fill the gap in the CloudFormation coverage. In this case you can use the AwsCustomResource construct. This construct creates a custom resource that can be customized to make specific API calls for the CREATE, UPDATE and DELETE events. Additionally, data returned by the API call can be extracted and used in other constructs/resources (creating a real CloudFormation dependency using Fn::GetAtt under the hood).

> The physical id of the custom resource can be specified or derived from the data returned by the API call.

The scenarios is that in a development environment I want to remove any CDK generated secrets and KMS keys.

My understanding is that I can use an AwsCustomResource with just the onDelete event specified, however creating the resource produces an Invalid PhysicalResourceId error.

### Reproduction Steps
```
  // Example
  const key = new Key(this, 'kms-key', {
      alias: 'KMS-Test-Delete-Alias',
       description: 'This is a test KMS key using custom resource',
       enabled: true,
   });

  new AwsCustomResource(this, 'kmsDeleteKeyResource', {
             onDelete: {
                service: 'KMS',
                action: 'scheduleKeyDeletion',
                parameters: {
                    KeyId: key.keyArn,
                    PendingWindowInDays: 7
                },
             },
        });
```

### Error Log

```
5/8 | 14:40:35 | CREATE_COMPLETE      | AWS::KMS::Key         | kms-key (kmskey49FBC3B3)
5/8 | 14:40:37 | CREATE_IN_PROGRESS   | AWS::KMS::Alias       | kms-key/Alias (kmskeyAlias39245779)
5/8 | 14:40:37 | CREATE_IN_PROGRESS   | Custom::AWS           | kmsDeleteKeyResource/Resource/Default (kmsDeleteKeyResource97160D9C)
5/8 Currently in progress: kmskeyAlias39245779, kmsDeleteKeyResource97160D9C
6/8 | 14:41:31 | CREATE_FAILED        | Custom::AWS           | kmsDeleteKeyResource/Resource/Default (kmsDeleteKeyResource97160D9C) Invalid PhysicalResourceId
               new CustomResource (C:\ScratchPad - AWS\cdk\kms-delete-key-api\node_modules\@aws-cdk\aws-cloudformation\lib\custom-resource.ts:163:21)
               \_ new AwsCustomResource (C:\ScratchPad - AWS\cdk\kms-delete-key-api\node_modules\@aws-cdk\custom-resources\lib\aws-custom-resource\aws-custom-resource.ts:209:27)
               \_ new DemoStack (C:\ScratchPad - AWS\cdk\kms-delete-key-api\example\demo-stack.ts:24:9)
               \_ Object.<anonymous> (C:\ScratchPad - AWS\cdk\kms-delete-key-api\example\cdk.ts:6:17)

```

### Environment

  - **CLI Version      :** aws-cli/1.16.185 Python/3.7.3 Windows/10 botocore/1.12.175
  - **Framework Version:** aws-cdk@1.22.0
  - **OS               :** Windows 10 1803
  - **Language         :** Typescript

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by jogold**
Correctly find the default physical resource id.

Default to logical resource id for a create event with delete only call.

Fixes #6061

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts [hunk 1]
```diff
@@ -86,7 +86,20 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
...
 
-    let physicalResourceId = (event as any).PhysicalResourceId;
+    // Default physical resource id
+    let physicalResourceId: string;
+    switch (event.RequestType) {
+      case 'Create':
+        physicalResourceId = event.ResourceProperties.Create?.physicalResourceId ??
+                             event.ResourceProperties.Update?.physicalResourceId ??
+                             event.ResourceProperties.Delete?.physicalResourceId ??
+                             event.LogicalResourceId;
```
Comment (jogold) [inline packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts]: Another option here is to make `physicalResourceId` mandatory also for the Delete call...

File: packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts [hunk 2]
```diff
@@ -86,7 +86,20 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
...
+        break;
+      case 'Update':
+      case 'Delete':
+        physicalResourceId = event.ResourceProperties[event.RequestType]?.physicalResourceId ?? event.PhysicalResourceId;
+    }
```
Comment (eladb) [inline packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts]:
```suggestion
      break;
    }
```

File: packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts (hunk 1 above)
Comment (eladb) [reply packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts]: I am good with this.

**Comment by jogold** [comment]
Can you remove the `pr-status/do-not-merge` label?

**Comment by jogold** [comment]
@eladb can you re-approve here?

**Pull request merged**